### PR TITLE
Update macaw-aarch32 for changes to the ASL translator

### PR DESCRIPTION
**Pull request by danmatichuk**

### CODE REVIEW

File: macaw-semmc/src/Data/Macaw/SemMC/TH.hs [hunk 1]
```diff
@@ -971,14 +986,14 @@ defaultAppEvaluator endianness elt interps = case elt of
...
             -> BoundVarInterpretations arch t fs
             -> BoolMapOp
             -> Bool
             -> BooM.BoolMap (S.Expr t)
             -> MacawQ arch t fs BoundExp
-evalBoolMap endianness interps op defVal bmap =
+_evalBoolMap endianness interps op defVal bmap =
   case BooM.viewBoolMap bmap of
     BooM.BoolMapUnit ->     letTH [| G.addExpr (boolBase $(lift defVal)) |]
     BooM.BoolMapDualUnit -> letTH [| G.addExpr (bNotBase $(lift defVal)) |]
```
Comment (travitch) [inline macaw-semmc/src/Data/Macaw/SemMC/TH.hs:999]: Are these now unused?

File: macaw-semmc/src/Data/Macaw/SemMC/TH/Monad.hs [hunk 2]
```diff
@@ -19,14 +19,27 @@ module Data.Macaw.SemMC.TH.Monad (
   letBindExpr,
   letBindPureExpr,
   bindTH,
+  letBindPat,
   letTH,
   extractBound,
   refBinding,
   inConditionalContext,
```
Comment (travitch) [inline macaw-semmc/src/Data/Macaw/SemMC/TH/Monad.hs:26]: Is this now unused? I think I saw the only use I recalled being removed earlier

File: macaw-aarch32/src/Data/Macaw/ARM/Semantics/TH.hs [hunk 3]
```diff
@@ -323,79 +350,42 @@ armNonceAppEval bvi nonceApp =
...
+                appendStmt $ [| join (execWriteAction <$> $(refBinding mem')) |]
+                liftQ [| return $ M.BoolValue True |]
 
+          "uf_noop" | Ctx.Empty <- args -> Just $ liftQ [| return $ M.BoolValue True |]
+
+          "uf_join_units"
+            | Ctx.Empty Ctx.:> u1 Ctx.:> u2 <- args -> Just $ do
+                _ <- addEltTH M.LittleEndian bvi u1
+                _ <- addEltTH M.LittleEndian bvi u2
+                liftQ [| return $ M.BoolValue True |]
```
[the remark this answers is not in the document]
Comment (danmatichuk) [reply macaw-aarch32/src/Data/Macaw/ARM/Semantics/TH.hs:388]: Sort of, this is only seen at the toplevel for instructions, and basically binds together the GPR update and SIMD update functions. Each of those individually will end up unwrapping the corresponding ARMWriteAction and execute it.

File: macaw-semmc/src/Data/Macaw/SemMC/TH.hs (hunk 1 above)
Comment (danmatichuk) [reply macaw-semmc/src/Data/Macaw/SemMC/TH.hs:999]: Yes, now the `BoolMap` in the `ConjPred` case gets turned into a list via `boolMapList` and then interpreted at runtime by either `allPreds` or `joinPreds`.

Comment (travitch) [reply macaw-semmc/src/Data/Macaw/SemMC/TH.hs:999]: Can we just delete the unused code then?

File: macaw-semmc/src/Data/Macaw/SemMC/TH/Monad.hs (hunk 2 above)
Comment (danmatichuk) [reply macaw-semmc/src/Data/Macaw/SemMC/TH/Monad.hs:26]: No, this is used in ARM.Semantics.TH to destruct a tuple in the `StructField` case of `armAppEvaluator`

**Pull request merged**